File: AndroidStudioProjects/DaggerDemoJava/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    Car car;
    @Inject
    Car car1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        CarComponent carComponent= ((ExampleApplication)getApplication()).getCarComponent();
        carComponent.inject(this);

        car.start();
        car1.start();
    }
}

File: AndroidStudioProjects/DaggerDemoKotlin/app/src/main/java/com/example/daggerdemokotlin/MainActivity.kt
package com.example.daggerdemokotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var car: Car
    @Inject
    lateinit var car1:Car

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var carComponent= (application as ExampleApp).carComponent()
        carComponent.inject(this)

        car.start()
        car1.start()
    }
}
